search-provider/eks-discovery-feed-provider.c

```diff
@@ -946,21 +967,25 @@ handle_get_videos (EksDiscoveryFeedProvider *skeleton,
     DmEngine *engine = dm_engine_get_default ();
     const char *tags_match_any[] = { "EknMediaObject", NULL };
 
-    /* Create query and run it */
-    g_autoptr(DmQuery) query = g_object_new (DM_TYPE_QUERY,
-                                             "tags-match-any", tags_match_any,
-                                             "sort", DM_QUERY_SORT_DATE,
-                                             "order", DM_QUERY_ORDER_DESCENDING,
-                                             "limit", SENSIBLE_QUERY_LIMIT,
-                                             "app-id", self->application_id,
-                                             NULL);
-
     /* Hold the application so that it doesn't go away whilst we're handling
      * the query */
     g_application_hold (g_application_get_default ());
 
-    dm_engine_query (engine, query, self->cancellable, relevant_video_cb,
-                     discovery_feed_query_state_new (invocation, self));
+    /* Create query and run it */
+    query_with_wraparound_offset (engine,
+                                  g_object_new (DM_TYPE_QUERY,
+                                                "content-type", "video",
+                                                "tags-match-any", tags_match_any,
+                                                "limit", 1,
```

Comment:
I realized this after merging the PR, but why do you use a limit of `1` here instead of `NUMBER_OF_ARTICLES`? I also see that the code in master/v3/v2 still manually counts videos in the callback, which is needed in the v1 code, but shouldn't be needed here. Am I missing something?